### ISSUE TITLE
Browser crash when "`GAYRİİHTİYARİ`" entered in Address Tab.

### ISSUE THREAD

**Issue by ycd** [brave/brave-browser#29348]
## Steps to Reproduce

   1.  Open Brave
   2. Search "`GAYRİİHTİYARİ`" through the Address Tab.
   3. Should crash.
   
## Reproduces how often: 

Everytime.

## Brave version (brave://version info)

Brave Version: v1.49.120
Device: Apple M1 13.2 Ventura

**Comment by stephendonner** [comment]
Thanks for the report, @ycd !  We'll see what we can do from Brave and/or Chromium's side.

**Comment by stephendonner** [comment]
https://share.backtrace.io/api/share/FS8Pd3LV39q03pibSDOTj0

```
[ 00 ] 0x7fff2042491e
[ 01 ] 0x7fff203a8406
[ 02 ] std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>>::basic_string(std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>> const&, unsigned long, unsigned long, std::Cr::allocator<char16_t> const&) ( string:0 )
[ 03 ] (anonymous namespace)::ExpandToFullWord(std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>> const&, std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>> const&) ( string:2238 )
[ 04 ] ShortcutsBackend::AddOrUpdateShortcut(std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>> const&, AutocompleteMatch const&) ( shortcuts_backend.cc:286 )
[ 05 ] ChromeOmniboxClient::OnSuccessfulNavigation(Profile*, std::Cr::basic_string<char16_t, std::Cr::char_traits<char16_t>, std::Cr::allocator<char16_t>> const&, AutocompleteMatch const&) ( chrome_omnibox_client.cc:511 )
[ 06 ] ChromeOmniboxNavigationObserver::DidFinishNavigation(content::NavigationHandle*) ( chrome_omnibox_navigation_observer.cc:295 )
[ 07 ] void content::WebContentsImpl::WebContentsObserverList::NotifyObservers<void (content::WebContentsObserver::*)(content::NavigationHandle*), content::NavigationHandle*&>(void (content::WebContentsObserver::*)(content::NavigationHandle*), content::NavigationHandle*&) ( web_contents_impl.h:1539 )
[ 08 ] content::WebContentsImpl::DidFinishNavigation(content::NavigationHandle*) ( web_contents_impl.cc:5815 )
[ 09 ] non-virtual thunk to content::WebContentsImpl::DidFinishNavigation(content::NavigationHandle*) ( web_contents_impl.cc:0 )
[ 10 ] content::NavigationRequest::~NavigationRequest() ( navigation_request.cc:2031 )
[ 11 ] content::NavigationRequest::~NavigationRequest() ( navigation_request.cc:1968 )
[ 12 ] content::NavigationRequest::~NavigationRequest() ( navigation_request.cc:1968 )
[ 13 ] std::Cr::default_delete<content::NavigationRequest>::operator()[abi:v170000](content::NavigationRequest*) const ( unique_ptr.h:65 )
[ 14 ] std::Cr::unique_ptr<content::NavigationRequest, std::Cr::default_delete<content::NavigationRequest>>::reset[abi:v170000](content::NavigationRequest*) ( unique_ptr.h:297 )
[ 15 ] content::Navigator::DidNavigate(content::RenderFrameHostImpl*, content::mojom::DidCommitProvisionalLoadParams const&, std::Cr::unique_ptr<content::NavigationRequest, std::Cr::default_delete<content::NavigationRequest>>, bool) ( navigator.cc:682 )
[ 16 ] content::RenderFrameHostImpl::DidCommitNavigationInternal(std::Cr::unique_ptr<content::NavigationRequest, std::Cr::default_delete<content::NavigationRequest>>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::InlinedStructPtr<content::mojom::DidCommitSameDocumentNavigationParams>) ( render_frame_host_impl.cc:12009 )
[ 17 ] content::RenderFrameHostImpl::DidCommitNavigation(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>) ( render_frame_host_impl.cc:12584 )
[ 18 ] void base::internal::FunctorTraits<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), void>::Invoke<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), content::RenderFrameHostImpl*, content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>>(void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), content::RenderFrameHostImpl*&&, content::NavigationRequest*&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>&&) ( bind_internal.h:745 )
[ 19 ] void base::internal::InvokeHelper<false, void, 0ul, 1ul>::MakeItSo<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), std::Cr::tuple<base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>>(void (content::RenderFrameHostImpl::*&&)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), std::Cr::tuple<base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>&&) ( bind_internal.h:924 )
[ 20 ] void base::internal::Invoker<base::internal::BindState<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>, void (mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>)>::RunImpl<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), std::Cr::tuple<base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>, 0ul, 1ul>(void (content::RenderFrameHostImpl::*&&)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), std::Cr::tuple<base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>&&, std::Cr::integer_sequence<unsigned long, 0ul, 1ul>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>&&) ( bind_internal.h:1019 )
[ 21 ] base::internal::Invoker<base::internal::BindState<void (content::RenderFrameHostImpl::*)(content::NavigationRequest*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>), base::internal::UnretainedWrapper<content::RenderFrameHostImpl, base::unretained_traits::MayNotDangle>, base::internal::UnretainedWrapper<content::NavigationRequest, base::unretained_traits::MayNotDangle>>, void (mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>)>::RunOnce(base::internal::BindStateBase*, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>&&, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>&&) ( bind_internal.h:970 )
[ 22 ] base::OnceCallback<void (mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>)>::Run(mojo::StructPtr<content::mojom::DidCommitProvisionalLoadParams>, mojo::StructPtr<content::mojom::DidCommitProvisionalLoadInterfaceParams>) && ( callback.h:152 )
[ 23 ] content::mojom::NavigationClient_CommitNavigation_ForwardToCallback::Accept(mojo::Message*) ( navigation_client.mojom.cc:1136 )
[ 24 ] mojo::InterfaceEndpointClient::HandleValidatedMessage(mojo::Message*) ( interface_endpoint_client.cc:1002 )
[ 25 ] mojo::MessageDispatcher::Accept(mojo::Message*) ( message_dispatcher.cc:43 )
[ 26 ] mojo::InterfaceEndpointClient::HandleIncomingMessage(mojo::Message*) ( interface_endpoint_client.cc:694 )
[ 27 ] void base::internal::FunctorTraits<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), void>::Invoke<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>(void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>&&, mojo::Message&&) ( ipc_mojo_bootstrap.cc:1075 )
[ 28 ] void base::internal::InvokeHelper<false, void, 0ul, 1ul>::MakeItSo<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), std::Cr::tuple<scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>>(void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*&&)(mojo::Message), std::Cr::tuple<scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>&&) ( bind_internal.h:924 )
[ 29 ] void base::internal::Invoker<base::internal::BindState<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>, void ()>::RunImpl<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), std::Cr::tuple<scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>, 0ul, 1ul>(void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*&&)(mojo::Message), std::Cr::tuple<scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>&&, std::Cr::integer_sequence<unsigned long, 0ul, 1ul>) ( bind_internal.h:1019 )
[ 30 ] base::internal::Invoker<base::internal::BindState<void (IPC::(anonymous namespace)::ChannelAssociatedGroupController::*)(mojo::Message), scoped_refptr<IPC::(anonymous namespace)::ChannelAssociatedGroupController>, mojo::Message>, void ()>::RunOnce(base::internal::BindStateBase*) ( bind_internal.h:970 )
[ 31 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::LazyNow*) ( callback.h:152 )
[ 32 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() ( thread_controller_with_message_pump_impl.cc:335 )
[ 33 ] non-virtual thunk to base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() ( thread_controller_with_message_pump_impl.cc:0 )
[ 34 ] base::MessagePumpCFRunLoopBase::RunWork() ( message_pump_mac.mm:475 )
[ 35 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 36 ] base::MessagePumpCFRunLoopBase::RunWorkSource(void*) ( message_pump_mac.mm:447 )
[ 37 ] 0x7fff2054bd2c
[ 38 ] 0x7fff2054bc94
[ 39 ] 0x7fff2054ba14
[ 40 ] 0x7fff2054a43c
[ 41 ] 0x7fff205499fc
[ 42 ] 0x7fff287931f3
[ 43 ] 0x7fff28792f55
[ 44 ] 0x7fff28792cf3
[ 45 ] 0x7fff22d52b12
[ 46 ] 0x7fff22d512e5
[ 47 ] __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke ( chrome_browser_application_mac.mm:239 )
[ 48 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 49 ] -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:] ( chrome_browser_application_mac.mm:238 )
[ 50 ] 0x7fff22d43609
[ 51 ] base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) ( message_pump_mac.mm:860 )
[ 52 ] base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) ( message_pump_mac.mm:172 )
[ 53 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) ( thread_controller_with_message_pump_impl.cc:644 )
[ 54 ] non-virtual thunk to base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) ( thread_controller_with_message_pump_impl.cc:0 )
[ 55 ] base::RunLoop::Run(base::Location const&) ( run_loop.cc:140 )
[ 56 ] content::BrowserMainLoop::RunMainMessageLoop() ( browser_main_loop.cc:1066 )
[ 57 ] content::BrowserMainRunnerImpl::Run() ( browser_main_runner_impl.cc:163 )
[ 58 ] content::BrowserMain(content::MainFunctionParams) ( browser_main.cc:32 )
[ 59 ] content::RunBrowserProcessMain(content::MainFunctionParams, content::ContentMainDelegate*) ( content_main_runner_impl.cc:715 )
[ 60 ] content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams, bool) ( content_main_runner_impl.cc:1263 )
[ 61 ] content::ContentMainRunnerImpl::Run() ( content_main_runner_impl.cc:1117 )
[ 62 ] content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) ( content_main.cc:321 )
[ 63 ] content::ContentMain(content::ContentMainParams) ( content_main.cc:349 )
[ 64 ] ChromeMain ( chrome_main.cc:180 )
[ 65 ] main ( chrome_exe_main_mac.cc:216 )
[ 66 ] 0x7fff2046ef3d
```

cc @brave/qa-team @rebron @bsclifton just FYI

**Comment by stephendonner** [comment]
@ycd mind confirming if https://github.com/brave/brave-browser/releases/tag/v1.49.132 fixes it for you?  Thanks!

**Comment by ycd** [comment]
I'm able to reproduce it on `v1.49.132` @stephendonner.

```
Brave: 1.49.132 Chromium: 111.0.5563.147 (Official Build) (arm64)

Revision: 029fd65a4dd711d1078468ed2206999cec86066b-refs/branch-heads/5563@{#1239}
OS: macOS Version 13.2.1 (Build 22D68)
```

**Comment by stephendonner** [comment]
> I'm able to reproduce it on `v1.49.132` @stephendonner.
> 
> ```
> Brave: 1.49.132 Chromium: 111.0.5563.147 (Official Build) (arm64)
> 
> Revision: 029fd65a4dd711d1078468ed2206999cec86066b-refs/branch-heads/5563@{#1239}
> OS: macOS Version 13.2.1 (Build 22D68)
> ```

Sorry, yeah, same, now - got into a state and it's nearly 100% :-(

We'll escalate this internally and see what we can do to mitigate.

**Comment by ycd** [comment]
Hey @stephendonner, something similar happened today, so I though adding more context would be helpful!

I was able to reproduce the crash while I was searching for some address, example:

```
Esentepe Mah, Büyükdere Cd.  No:209, 34394 Şişli/İstanbul
```


So this got me thinking because my search had something similar to my previous _input_ that was causing the crash. Then I changed `İstanbul` to `istanbul` while keeping all the address before, it did not crash.

```
Esentepe Mah, Büyükdere Cd.  No:209, 34394 Şişli/istanbul
```

The address is one of the biggest corporate banks in Turkey so I think the reproducibility of the bug is much higher than I expected and AFAICT the bug is related to Chromium, should I cross-post this to bug tracker?
